sharding/nodesCoordinator/indexHashedNodesCoordinator.go

```diff
@@ -226,10 +229,46 @@ func checkArguments(args ArgNodesCoordinator) error {
 	if args.NumStoredEpochs < minStoredEpochs {
 		return ErrInvalidNumberOfStoredEpochs
 	}
+	if check.IfNilReflect(args.NodesConfigCache) {
+		return ErrNilNodesConfigCacher
+	}
 
 	return nil
 }
 
+// getNodesConfig will try to get nodesConfig from map, if it doesn't success, it will try to get it from nodes config cache
+// it has to be used under mutex
+func (ihnc *indexHashedNodesCoordinator) getNodesConfig(epoch uint32) (*epochNodesConfig, bool) {
+	nodesConfig, ok := ihnc.nodesConfig[epoch]
+	if ok {
+		return nodesConfig, ok
+	}
+
+	value, ok := ihnc.nodesConfigCacher.Get([]byte(fmt.Sprint(epoch)))
```

Comment:
I think we can add in the cache the object directly unmarshalled so we don't need to unmarshal it if found.
if not found, we could look into the nodes coordinator registry storer (bootstrap data storage I think) with a GetFromEpoch, and if found unmarshal it and add it to the LRU cache.

then maybe rename also the method to getNodesConfigWithCacheUpdate

Reply:
Agree, Cachers can hold just about everything as value

Reply:
done